src/openforms/forms/api/viewsets.py

```diff
@@ -517,3 +521,47 @@ def post(self, request, *args, **kwargs):
         import_form(serializer.validated_data["file"])
 
         return response.Response(status=status.HTTP_204_NO_CONTENT)
+
+
+@extend_schema_view(
+    bulk_update=extend_schema(
+        summary=_("Bulk update/create/delete form variables"),
+        tags=["forms"],
+        parameters=[
+            OpenApiParameter(
+                name="form_uuid_or_slug",
+                location=OpenApiParameter.PATH,
+                type=str,
+                description=_("Either a UUID4 or a slug identifying the form."),
+            ),
+        ],
+        request=FormVariableListSerializer,
+        responses={status.HTTP_201_CREATED: FormVariableListSerializer},
+    ),
+)
+class FormVariablesViewSet(NestedViewSetMixin, viewsets.GenericViewSet):
+    permission_classes = [FormAPIPermissions]
+    serializer_class = FormVariableSerializer
+    queryset = FormVariable.objects.all()
+    parent_lookup_kwargs = {"form_uuid_or_slug": "form__uuid"}
+
+    def initialize_request(self, request, *args, **kwargs):
+        # Skip the NestedViewSetMixin initialize request which would place the form URL in the request data
+        # It doesn't support the data being a list of dicts instead of a dict
+        return super(_NestedViewSetMixin, self).initialize_request(
+            request, *args, **kwargs
+        )
+
+    @action(detail=False, methods=["put"])
+    @transaction.atomic
+    def bulk_update(self, request, *args, **kwargs):
+        form_variables = self.get_queryset()
+        # We expect that all the variables that should be associated with a form come in the request.
+        # So we can delete any existing variables because they will be replaced.
+        form_variables.delete()
+
+        serializer = self.get_serializer(data=request.data, many=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
```

Comment:
will have to be a HTTP_200, the 201 semantics really only apply to a single resource rather than a collection of resources (for example, DRF emits the `Location` header of the newly created object, which doesn't make any sense here)